lib/src/samples/authenticate_with_oauth_sample.dart

```diff
@@ -0,0 +1,96 @@
+import 'package:flutter/material.dart';
+import 'package:arcgis_maps/arcgis_maps.dart';
+
+class AuthenticateWithOAuthSample extends StatefulWidget {
+  const AuthenticateWithOAuthSample({
+    super.key,
+    required this.title,
+  });
+  final String title;
+
+  @override
+  State<AuthenticateWithOAuthSample> createState() =>
+      _AuthenticateWithOAuthSampleState();
+}
+
+class _AuthenticateWithOAuthSampleState
+    extends State<AuthenticateWithOAuthSample>
+    implements ArcGISAuthenticationChallengeHandler {
+  // This document describes the steps to configure OAuth for your app:
+  // https://developers.arcgis.com/documentation/mapping-apis-and-services/security/user-authentication/serverless-native-flow/
+  final _oauthUserConfiguration = OAuthUserConfiguration(
+    portalUri: Uri.parse('https://www.arcgis.com'),
+    clientId: 'lgAdHkYZYlwwfAhC',
+    redirectUri: Uri.parse('my-ags-app://auth'),
+  );
+  final _mapViewController = ArcGISMapView.createController();
+
+  @override
+  void initState() {
+    super.initState();
+
+    // This class implements the ArcGISAuthenticationChallengeHandler interface,
+    // which allows it to handle authentication challenges via calls to its
+    // handleArcGISAuthenticationChallenge() method.
+    ArcGISEnvironment
+        .authenticationManager.arcGISAuthenticationChallengeHandler = this;
+
+    // A portal item map that has a secure layer (traffic).
+    _mapViewController.arcGISMap = ArcGISMap.withItem(
+      PortalItem.withPortalAndItemId(
+        portal: Portal.arcGISOnline(connection: PortalConnection.authenticated),
+        itemId: 'e5039444ef3c48b8a8fdc9227f9be7c1',
+      ),
+    );
+  }
+
+  @override
+  void dispose() async {
+    super.dispose();
+
+    ArcGISEnvironment
+        .authenticationManager.arcGISAuthenticationChallengeHandler = null;
```

Comment:
Maybe a comment for this step too? Just to emphasize it as part of the workflow.

Reply:
added